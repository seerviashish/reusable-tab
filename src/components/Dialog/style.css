.dialog-form {
  color: #423e3e;
  font-size: 1.3rem;
  font-weight: 500;
  box-sizing: border-box;
  display: flex;
  flex-flow: column;
  max-width: 250px;
}

h6 {
  margin: 16px 0;
}

label {
  display: flex;
  color: grey;
  font-weight: 500;
}
input {
  width: 100%;
}

button {
  margin-top: 16px;
  margin-right: 16px;
}
