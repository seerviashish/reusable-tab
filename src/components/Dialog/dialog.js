import React from "react";
import "./style.css";

const Dialog = (props) => {
  const { title, onCancel, onCreate } = props;
  const [tab, setTab] = React.useState("");
  const handleInput = (e) => {
    setTab(e.target.value);
  };
  const handleOnCreate = (e) => {
    onCreate(tab);
  };
  return (
    <dialog className="dialog-form">
      <h6>{title}</h6>
      <div>
        <label htmlFor="tabName">Enter tab name.</label>
        <input
          type="text"
          id="tabName"
          name="tabName"
          value={tab}
          onChange={handleInput}
        />
        <button onClick={onCancel}>Cancel</button>
        <button onClick={handleOnCreate}>Create</button>
      </div>
    </dialog>
  );
};

export default Dialog;
