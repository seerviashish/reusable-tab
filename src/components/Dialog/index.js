import React from "react";
import "./style.css";

const Dialog = (props) => {
  const { title, onCancel, onCreate, type, warnMessage } = props;
  const [tab, setTab] = React.useState("");
  const handleInput = (e) => {
    setTab(e.target.value);
  };
  const handleOnCreate = (e) => {
    onCreate(tab);
  };
  return (
    <dialog className="dialog-form" data-testid="dialog-form">
      <h6 data-testid="dialog-title">{title}</h6>
      <div>
        {type === "form" && (
          <>
            <label data-testid="label" htmlFor="tabName">
              Enter tab name.
            </label>
            <input
              data-testid="input"
              type="text"
              id="tabName"
              name="tabName"
              value={tab}
              onChange={handleInput}
            />
          </>
        )}
        {type !== "form" && (
          <p data-testid="warn-message" style={{ color: "red" }}>
            {warnMessage}
          </p>
        )}
        <button data-testid="cancel-btn" onClick={onCancel}>
          {type === "form" ? "Cancel" : "Close"}
        </button>
        {type === "form" && (
          <button data-testid="create-btn" onClick={handleOnCreate}>
            Create
          </button>
        )}
      </div>
    </dialog>
  );
};

export default Dialog;
