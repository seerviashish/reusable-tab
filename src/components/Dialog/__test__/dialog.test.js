import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import Dialog from "../index";
import pretty from "pretty";
import { act } from "@testing-library/react";

let container = null;

describe("Dialog Component Test", () => {
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("should render dialog warning message.", () => {
    const handleDialogClose = (e) => {};
    const warnTitle = "Warning!";
    const warningMessage = "This is warning message!";
    act(() => {
      render(
        <Dialog
          title={warnTitle}
          type="warn"
          onCancel={handleDialogClose}
          warnMessage={warningMessage}
        />,
        container
      );
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
    expect(
      container.querySelector("[data-testid='dialog-title']")
    ).toHaveTextContent(warnTitle);
    expect(
      container.querySelector("[data-testid='warn-message']")
    ).toHaveTextContent(warningMessage);
    expect(
      container.querySelector("[data-testid='cancel-btn']")
    ).toHaveTextContent("Close");
    expect(container.querySelector("[data-testid='create-btn']")).toBeNull();
  });

  it("should render dialog dialog form.", () => {
    const handleDialogClose = (e) => {
      return true;
    };
    const handleDialogCreate = (tabName) => {
      return tabName;
    };
    const dialogTitle = "Add Tab";
    act(() => {
      render(
        <Dialog
          title={dialogTitle}
          type="form"
          onCancel={handleDialogClose}
          onCreate={handleDialogCreate}
        />,
        container
      );
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
    expect(
      container.querySelector("[data-testid='dialog-title']")
    ).toHaveTextContent(dialogTitle);
    expect(container.querySelector("[data-testid='label']")).toHaveTextContent(
      "Enter tab name."
    );
    expect(container.querySelector("[data-testid='input']")).toHaveAttribute(
      "type",
      "text"
    );
    expect(container.querySelector("[data-testid='input']")).toHaveAttribute(
      "id",
      "tabName"
    );
    expect(container.querySelector("[data-testid='input']")).toHaveAttribute(
      "name",
      "tabName"
    );
    expect(
      container.querySelector("[data-testid='cancel-btn']")
    ).toHaveTextContent("Cancel");
    expect(
      container.querySelector("[data-testid='create-btn']")
    ).toHaveTextContent("Create");
    expect(container.querySelector("[data-testid='warn-message']")).toBeNull();
  });
});
