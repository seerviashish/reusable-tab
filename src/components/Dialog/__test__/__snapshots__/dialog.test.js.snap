// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component Test should render dialog dialog form. 1`] = `
"<dialog class=\\"dialog-form\\" data-testid=\\"dialog-form\\">
  <h6 data-testid=\\"dialog-title\\">Add Tab</h6>
  <div><label data-testid=\\"label\\" for=\\"tabName\\">Enter tab name.</label><input data-testid=\\"input\\" type=\\"text\\" id=\\"tabName\\" name=\\"tabName\\" value=\\"\\"><button data-testid=\\"cancel-btn\\">Cancel</button><button data-testid=\\"create-btn\\">Create</button></div>
</dialog>"
`;

exports[`Dialog Component Test should render dialog warning message. 1`] = `
"<dialog class=\\"dialog-form\\" data-testid=\\"dialog-form\\">
  <h6 data-testid=\\"dialog-title\\">Warning!</h6>
  <div>
    <p data-testid=\\"warn-message\\" style=\\"color: red;\\">This is warning message!</p><button data-testid=\\"cancel-btn\\">Close</button>
  </div>
</dialog>"
`;
